lib/Test/Nginx/Socket.pm

```diff
@@ -611,7 +611,11 @@ sub run_test_helper ($$) {
         $cmd = quote_sh_args($cmd);
 
         warn "$cmd\n";
-        system "unbuffer $cmd > /dev/stderr";
+        my $full_cmd = "unbuffer $cmd > /dev/stderr";
+        my $rc = system $full_cmd;
+        if ($rc != 0) {
+          bail_out("Failure running '${full_cmd}': return code ${rc}");
```

Comment:
You can use the special variable `$?` here for the error code. Also, better to use "Failed to run '...'" as the error message. And the curly braces for the `$rc` reference is not necessary.